### PR TITLE
fix(executor): Increase pod patch backoff. Fixes #4339

### LINKED ISSUE #4339: Slow / Unresponsive Kubernetes API
## Summary 
Not sure if this is Argo or AWS Kubernetes related... I would label it more as a "Problem" than a bug.

If we run ~300 Argo workflows (about 8 non parallel steps each) in parallel on our EKS Cluster which takes ~70 Nodes the Kubernetes API starts to get slow and in worst case we experience API timeouts.
The worst thing is that it also makes the workflows fail with different error messages:
```
failed to save outputs: the server was unable to return a response in the time allotted, but may still be processing the request (patch pods bx59k-552586920)

failed to save outputs: Patch https://10.100.0.1:443/api/v1/namespaces/v1-0/pods/5r49l-2608568875: net/http: TLS handshake timeout

failed to save outputs: Patch https://10.100.0.1:443/api/v1/namespaces/v1-0/pods/nx5b7-2156588813: http2: server sent GOAWAY and closed the connection; LastStreamID=5, ErrCode=NO_ERROR, debug=""

failed to save outputs: Patch https://10.100.0.1:443/api/v1/namespaces/v1-0/pods/9q96d-4078545465: stream error: stream ID 5; INTERNAL_ERROR

failed to save outputs: Patch https://10.100.0.1:443/api/v1/namespaces/v1-0/pods/ztglr-1531658454: dial tcp 10.100.0.1:443: i/o timeout

failed to save outputs: Patch https://10.100.0.1:443/api/v1/namespaces/v1-0/pods/wg9qh-2738839668: unexpected EOF
```
We have scaled up the workflow-controller with `--workflow-workers 1024 --pod-workers 64 --qps 200 --burst 50`.
We would expect Argo / Kubernetes to handle such workload without breaking.

## Diagnostics
We are using AWS EKS 1.17 and Argo 2.11.5

```
time="2020-10-21T15:37:22Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-svc-gatk-mutect-A-v1-chr-11-pair-X-v1-c9z2g
time="2020-10-21T15:37:22Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-svc-annotation-B-v1-pair-1a2166-tn7-v1-fnhn4
time="2020-10-21T15:37:22Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-svc-annotation-B-v1-pair-1a2166-tn7-v1-fnhn4
time="2020-10-21T15:37:22Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-post-alignment-C-v1-hq4sj
time="2020-10-21T15:37:22Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-svc-annotation-B-v1-pair-1a2166-tn7-v1-fnhn4
time="2020-10-21T15:37:22Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-alignment-Z-v1-lane-hyltldsxx-003-lp67f
time="2020-10-21T15:37:22Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-alignment-Z-v1-lane-hyltldsxx-003-lp67f
time="2020-10-21T15:37:22Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-alignment-Z-v1-lane-hyltldsxx-003-lp67f
time="2020-10-21T15:37:22Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-D-v1-5gzg6
time="2020-10-21T15:37:22Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-D-v1-5gzg6
time="2020-10-21T15:37:22Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-post-alignment-D-v1-5gzg6
time="2020-10-21T15:37:23Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-alignment-E-v1-lane-hyllvdsxx-003-hzljh
time="2020-10-21T15:37:23Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-alignment-E-v1-lane-hyllvdsxx-003-hzljh
time="2020-10-21T15:37:23Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-alignment-E-v1-lane-hyllvdsxx-003-hzljh
time="2020-10-21T15:37:23Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-raw-data-F-v1-lwgnt
time="2020-10-21T15:37:23Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-raw-data-F-v1-lwgnt
time="2020-10-21T15:37:23Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-raw-data-F-v1-lwgnt
time="2020-10-21T15:37:23Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-gvc-haplotype-caller-A-v1-chr-9-v9r5v
time="2020-10-21T15:37:23Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-gvc-haplotype-caller-A-v1-chr-9-v9r5v
time="2020-10-21T15:37:23Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-gvc-haplotype-caller-A-v1-chr-9-v9r5v
time="2020-10-21T15:37:24Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-G-v1-nl5bk
time="2020-10-21T15:37:24Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-G-v1-nl5bk
time="2020-10-21T15:37:24Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-post-alignment-G-v1-nl5bk
time="2020-10-21T15:37:24Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-H-v1-48r8r
time="2020-10-21T15:37:24Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-H-v1-48r8r
time="2020-10-21T15:37:24Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-post-alignment-H-v1-48r8r
time="2020-10-21T15:37:24Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-svc-annotation-I-v1-pair-Y-v1-88mb9
time="2020-10-21T15:37:24Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-svc-annotation-I-v1-pair-Y-v1-88mb9
time="2020-10-21T15:37:24Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-svc-annotation-I-v1-pair-Y-v1-88mb9
time="2020-10-21T15:37:24Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-svc-gatk-mutect-A-v1-chr-4-pair-X-v1-9lvk6
time="2020-10-21T15:37:24Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-svc-gatk-mutect-A-v1-chr-4-pair-X-v1-9lvk6
time="2020-10-21T15:37:24Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-svc-gatk-mutect-A-v1-chr-4-pair-X-v1-9lvk6
time="2020-10-21T15:37:24Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-gvc-haplotype-caller-A-v1-chr-19-2lmbt
time="2020-10-21T15:37:24Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-gvc-haplotype-caller-A-v1-chr-19-2lmbt
time="2020-10-21T15:37:24Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-gvc-haplotype-caller-A-v1-chr-19-2lmbt
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-gvc-annotation-A-v1-chr-x-8slcb
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-gvc-annotation-A-v1-chr-x-8slcb
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-alignment-1a437-tp1-v1-lane-hyltldsxx-002-vp4ch
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-alignment-1a437-tp1-v1-lane-hyltldsxx-002-vp4ch
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-gvc-annotation-A-v1-chr-x-8slcb
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-gvc-haplotype-caller-A-v1-chr-18-4wgpk
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-gvc-haplotype-caller-A-v1-chr-18-4wgpk
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-alignment-1a437-tp1-v1-lane-hyltldsxx-002-vp4ch
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-1e273-tp1-v1-wzmtk
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-1e273-tp1-v1-wzmtk
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-gvc-haplotype-caller-A-v1-chr-18-4wgpk
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-1d806-tp1-v1-tw528
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-1d806-tp1-v1-tw528
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-post-alignment-1e273-tp1-v1-wzmtk
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-post-alignment-1d806-tp1-v1-tw528
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-svc-varscan-A-v1-chr-10-pair-X-v1-w8lx2
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-svc-varscan-A-v1-chr-10-pair-X-v1-w8lx2
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-alignment-Z-v1-lane-hyltldsxx-002-kn977
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-alignment-Z-v1-lane-hyltldsxx-002-kn977
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-gvc-haplotype-caller-A-v1-chr-4-rr5wx
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-gvc-haplotype-caller-A-v1-chr-4-rr5wx
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-raw-data-qc-A-v1-47btt
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-raw-data-qc-A-v1-47btt
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-raw-data-qc-Y-v1-74pdr
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-raw-data-qc-Y-v1-74pdr
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-svc-varscan-A-v1-chr-10-pair-X-v1-w8lx2
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-svc-annotation-Z-v1-pair-1c682-tn7-v1-rmvhj
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-svc-annotation-Z-v1-pair-1c682-tn7-v1-rmvhj
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-raw-data-qc-A-v1-47btt
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-gvc-haplotype-caller-A-v1-chr-4-rr5wx
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-alignment-Z-v1-lane-hyltldsxx-002-kn977
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-raw-data-qc-Y-v1-74pdr
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-svc-annotation-Z-v1-pair-1c682-tn7-v1-rmvhj
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-Y-v1-xrwc8
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-Y-v1-xrwc8
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-post-alignment-Y-v1-xrwc8
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-alignment-U-v1-lane-h2gfwdsxy-003-kdj99
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-alignment-U-v1-lane-h2gfwdsxy-003-kdj99
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-svc-annotation-L-v1-pair-G-v1-gclvq
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-svc-annotation-L-v1-pair-G-v1-gclvq
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-alignment-O-v1-lane-h2ckjdsxy-003-54wxz
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-alignment-O-v1-lane-h2ckjdsxy-003-54wxz
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-svc-gatk-mutect-A-v1-chr-17-pair-X-v1-ptt9b
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-svc-gatk-mutect-A-v1-chr-17-pair-X-v1-ptt9b
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-svc-annotation-L-v1-pair-G-v1-gclvq
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-alignment-U-v1-lane-h2gfwdsxy-003-kdj99
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-svc-gatk-mutect-A-v1-chr-17-pair-X-v1-ptt9b
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-alignment-O-v1-lane-h2ckjdsxy-003-54wxz
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-K-v1-dbxm7
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-K-v1-dbxm7
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-post-alignment-K-v1-dbxm7
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-X-v1-dqsxd
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-X-v1-dqsxd
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-post-alignment-X-v1-dqsxd
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-Y-v1-p2kzg
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-post-alignment-Y-v1-p2kzg
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-post-alignment-Y-v1-p2kzg
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-raw-data-qc-M-v1-bllv4
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-raw-data-qc-M-v1-bllv4
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-raw-data-qc-M-v1-bllv4
time="2020-10-21T15:37:25Z" level=warning msg="Deadline exceeded" namespace=v1-0 workflow=wgs-svc-varscan-A-v1-chr-7-pair-X-v1-ztdq7
time="2020-10-21T15:37:25Z" level=error msg="error in entry template execution" error="Deadline exceeded" namespace=v1-0 workflow=wgs-svc-varscan-A-v1-chr-7-pair-X-v1-ztdq7
time="2020-10-21T15:37:25Z" level=info msg="Processing workflow" namespace=v1-0 workflow=wgs-svc-varscan-A-v1-chr-7-pair-X-v1-ztdq7

```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).
* [x] I've signed the CLA.
* [ ] ~~I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.~~
* [x] My builds are green. Try syncing with master if they are not. 

Fixes #4339

### CODE REVIEW

**Comment by TekTimmy** [comment]
It would be **awesome** if you could release this fix soon because its a real deal breaker for us currently. :+1:

File: workflow/executor/executor.go [hunk 1]
```diff
@@ -40,6 +40,12 @@ import (
...
 
 // ExecutorRetry is a retry backoff settings for WorkflowExecutor
+// Run	Seconds
+// 0	0.000
+// 1	1.000
+// 2	2.600
+// 3	5.160
+// 4	9.256
 var ExecutorRetry = wait.Backoff{
 	Steps:    5,
```
Comment (alexec) [inline workflow/executor/executor.go:50]: maybe even more steps?

File: workflow/common/util.go [hunk 2]
```diff
@@ -434,44 +435,54 @@ func RunShellCommand(arg ...string) ([]byte, error) {
...
+			panic("unknown option type")
+		}
+	}
+	return addPodMetadata(c, "annotations", podName, namespace, key, value, backoff)
 }
 
 // AddPodLabel adds an label to pod
 func AddPodLabel(c kubernetes.Interface, podName, namespace, key, value string) error {
-	return addPodMetadata(c, "labels", podName, namespace, key, value)
+	return addPodMetadata(c, "labels", podName, namespace, key, value, defaultPatchBackoff)
```
Comment (whynowy) [inline workflow/common/util.go:466]: What is the reason `defaultPatchBackoff` is used here but a passed in value is used in `AddPodAnnotations()`?

Comment (alexec) [reply workflow/common/util.go:466]: So that the API does not change.

**Review by whynowy — APPROVED** [review]
LGTM

**Pull request merged**